src/hooks/useMarketIndication.ts

```diff
@@ -0,0 +1,83 @@
+import { useMemo } from 'react';
+import BigNumber from 'bignumber.js';
+import { useGetTokenPrice } from 'libs/queries';
+import { Token } from 'libs/tokens';
+import { useFiatCurrency } from './useFiatCurrency';
+import { OrderCreate } from 'components/strategies/create/useOrder';
+
+export type MarketPricePercentage = {
+  min: BigNumber;
+  max: BigNumber;
+  price: BigNumber;
+};
+
+export type UseMarketIndicationProps = {
+  base: Token | undefined;
+  quote: Token | undefined;
+  order: OrderCreate;
+  buy?: boolean;
+};
+
+export const useMarketIndication = ({
+  base,
+  quote,
+  order,
+  buy = false,
+}: UseMarketIndicationProps) => {
+  const baseTokenPriceQuery = useGetTokenPrice(base?.address);
+  const { getFiatValue, selectedFiatCurrency } = useFiatCurrency(quote);
+  const tokenMarketPrice =
+    baseTokenPriceQuery?.data?.[selectedFiatCurrency] || 0;
+
+  const isOrderAboveOrBelowMarketPrice = useMemo(() => {
+    if (order.isRange) {
+      const isInputNotZero = buy
+        ? new BigNumber(order.max).gt(0)
+        : new BigNumber(order.min).gt(0);
+
+      return (
+        isInputNotZero &&
+        new BigNumber(getFiatValue(buy ? order.max : order.min))[
+          buy ? 'gt' : 'lt'
+        ](tokenMarketPrice)
```

Comment:
this is really hard to read and understand whats going on here, lets clean this up a bit.